_episodes/06-control-flow.md

```diff
@@ -165,17 +212,50 @@ and checks the new distance.
 ~~~
 julia> better_trebuchet = imprecise_trebuchet - 0.05 * grad;
 
-julia> shoot_distance([5, new_arguments...])
+julia> shoot_distance([5, better_trebuchet[2], better_trebuchet[1]])
 58.871526223121755
 ~~~
 {: .language-julia}
 
-That got shorter, but also a bit too short.
+Great! That didn't shoot past the target, but instead it landed a bit too short.
 
 > ## Experiment
 >
 > How far can you change the parameters in the direction of the gradient, such
 > that it still improves the distance?
+>
+> > ## Evaluation
+> >
+> > Try a bunch of values!
+> >
+> > * ~~~
+> >   better_trebuchet = imprecise_trebuchet - 0.04 * grad
+> >   shoot_distance([environment.wind, better_trebuchet[2], better_trebuchet[1]])
+> >   120.48753521261001
+> >   ~~~
+> >   {: .language-julia}
+> > * ~~~
+> >   better_trebuchet = imprecise_trebuchet - 0.03 * grad
+> >   shoot_distance([environment.wind, better_trebuchet[2], better_trebuchet[1]])
+> >   107.80646596787481
+> >   ~~~
+> >   {: .language-julia}
+> > * ~~~
+> >   better_trebuchet = imprecise_trebuchet - 0.02 * grad
+> >   shoot_distance([environment.wind, better_trebuchet[2], better_trebuchet[1]])
+> >   33.90699307740854
+> >   ~~~
+> >   {: .language-julia}
+> > * ~~~
+> >   better_trebuchet = imprecise_trebuchet - 0.025 * grad
+> >   shoot_distance([environment.wind, better_trebuchet[2], better_trebuchet[1]])
+> >   75.87613276409223
+> >   ~~~
+> >   {: .language-julia}
+> >
+> > Looks like the "best" trebuchet for a target 100 m away will be between
+> > 2.5% and 3% down the gradient from the imprecise trebuchet.
```

Comment:
Well, the take home message from this experiment should be, that this is a highly non-linear chaotic system, where you have little chance to get a useful value by trial-and-error.

Reply:
I disagree: that's the take-home of the following section. The released version of this "Experiment" is to see how close you can get, without providing any guidance to the instructor or self-directed learner. The introduced "Evaluation" shows how to tweak the trebuchet and (implicitly) by how much to achieve a better result: `η = [0.04, 0.03, 0.02, 0.025] -> d = [120, 107, 33, 75]`. This sequence gives the excitement of success (getting closer!) and encourages further experimentation to get a perfect answer, which happens to be what the very next section is all about. That's a better place to introduce the frustrations of nonlinear systems.

Reply:
Fair point